src/library.js

```diff
@@ -802,9 +802,11 @@ LibraryManager.library = {
     name = UTF8ToString(name);
     if (!ENV.hasOwnProperty(name)) return 0;
 
-    if (_getenv.ret) _free(_getenv.ret);
-    _getenv.ret = allocateUTF8(ENV[name]);
-    return _getenv.ret;
+    // Use 1 location for each name.  Use an ES6 map
+    if (!_getenv.ret) _getenv.ret = new Map()
+    if (_getenv.ret.has(name)) _free(_getenv.ret.get(name));
+    _getenv.ret.set(name, allocateUTF8(ENV[name]));
+    return _getenv.ret.get(name);
```

Comment:
I'm not sure about use of ES6 features like Map.  Emscripten currently targets ES5.  We'd like to change that one day but for an object is probably better.  

Also, I wonder if we can/should avoid the relocation if the string hasn't changed?  That way `getenv("FOO")` twice in row would only allocate once and return the same point each time.  Not essential but maybe a little tidier.

Reply:
I did wonder about ES6.  I was concerned an object could lead to weird bugs given the predefined fields on it.

Less allocation may be useful at the cost of an extra string comparison.